### DISCUSSION TITLE
Batch update

### DISCUSSION THREAD

**Discussion by Penkie** [pocketbase/pocketbase#843] [Q&A] [answered]
Hi everyone,

I have a question about batch update. I'm migrating my project from Firebase to Pocketbase and I'm wondering if there's any feature like this.

In short, I have a todo list and it has a priority field to order on the front-end my todos. Now I have a drag-n-drop feature where the user can just drag a todo and drop it somewhere else. When using Firebase, I did a batch update to update all the todos the are concerned by the change (only the ones with the same listId) to change the priority field with the current order of my array.
Is there any alternative to this in pocketbase ?
The only other solution I see is a good old foreach but I have have 20 todos it'll do 20 calls and that's not pretty...

Thanks in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
Currently there is no support for transactional bulk create/update/delete of records data via the API. It will be added in the near future as part of #48.

The only option right now is to insert each item with a separate request.

_Note that if you are using the [JavaScript SDK](https://github.com/pocketbase/js-sdk), you may have to disable autocancellation if you are not awaiting each individual create request. For an example, you could check https://github.com/pocketbase/pocketbase/discussions/824#discussioncomment-3942295._

**Reply by Penkie** [reply]
Okey, will wait this feature then. In the meantime gonna have to do a loooots of requests but that's okey if it's temporary !

**Comment by heyarviind** [comment]
this is something we need, can this be prioritized?

**Comment by thedittmer** [comment]
I agree. I need this too.

**Comment by eskutti** [comment]
Yes i do Agree, we need this feature ASAP.

**Comment by Christopher-Crescenzo-Imprivata** [comment]
Any progress on this feature?

**Reply by ganigeorgiev** [reply]
Nothing has changed here. See also https://github.com/pocketbase/pocketbase/discussions/4355.
